groth16-framework/test_data/Groth16VerifierExtensions.sol

```diff
@@ -136,54 +157,78 @@ contract Query2 is Verifier {
     }
 
     // Verify the plonky2 inputs with the expected Query instance.
-    function verifyQuery(bytes memory pis, Query memory query) internal pure {
-        uint32 min_block_number = convertToU32(pis, PI_MIN_BLOCK_NUM_OFFSET);
+    function verifyQuery(bytes memory pis, uint8 identifier, Query memory query) internal pure {
+        uint32 minBlockNumber = convertToU32(pis, PI_MIN_BLOCK_NUM_OFFSET);
         require(
-            min_block_number == query.minBlockNumber,
+            minBlockNumber == query.minBlockNumber,
             "The parsed min block number must be equal to the expected one in query."
         );
 
-        uint32 max_block_number = convertToU32(pis, PI_MAX_BLOCK_NUM_OFFSET);
+        uint32 maxBlockNumber = convertToU32(pis, PI_MAX_BLOCK_NUM_OFFSET);
         require(
-            max_block_number == query.maxBlockNumber,
+            maxBlockNumber == query.maxBlockNumber,
             "The parsed max block number must be equal to the expected one in query."
         );
 
-        address contract_address = convertToAddress(pis, PI_CONTRACT_ADDR_OFFSET);
+        address contractAddress = convertToAddress(pis, PI_CONTRACT_ADDR_OFFSET);
         require(
-            contract_address == query.contractAddress,
+            contractAddress == query.contractAddress,
             "The parsed contract address must be equal to the expected one in query."
         );
 
-        address user_address = convertToAddress(pis, PI_USER_ADDR_OFFSET);
+        address userAddress = convertToAddress(pis, PI_USER_ADDR_OFFSET);
```

Comment:
As discussed offline, I would add here (and thus also to the query inputs) checks over the mapping slot and the length slot. Wdyt? Tagging also @nikkolasg to reach an agreement on this.